### PR TITLE
[bitnami/all] docs(kubernetes): replace broken links to ./user-guide/ #17782

### LINKED ISSUE #17782: docs: links to https://kubernetes.io/docs/user-guide/...  end in 404 error
### Name and Version

bitnami/all

### What architecture are you using?

None

### What steps will reproduce the bug?

View a YAML files that includes a hyperlink, similar to the following:
```YAML
    ## ref: https://kubernetes.io/docs/user-guide/images/#pre-pulling-images
```

### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

Following a link loads the intended/expected content from kubernetes.io

### What do you see instead?

HTTP 404 error message from https://kubernetes.io/docs/ site

### Additional information

I intend to submit an MR to replace all ref links that include a link like `ref: https://kubernetes.io/docs/user-guide/` with a current/functional link

### PR DESCRIPTION
**Pull request by bcdady**
replace broken links to `./user-guide/` with current links to `./concepts/`; primarily in values.yaml files

with current links to ./concepts/

### Description of the change

Replace broken 'ref' links in values.yaml files with working links 

'Bad' links includes https://kubernetes.io/docs/user-guide/ in the path. Replacement links reference /concepts/ in the path instead.

### Benefits

Provides accurate reference for users of charts

### Applicable issues

- fixes #17782

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/appsmith/values.yaml
Comment (FraPazGal) [inline bitnami/appsmith/values.yaml:1]: It seems that the URLs related to the `pullPolicy` parameter were not changed in those `values.yaml` where a `persistence` param existed. Could you check this?

Comment (bcdady) [reply bitnami/appsmith/values.yaml:1]: Certainly. I'll look into it.

Comment (bcdady) [reply bitnami/appsmith/values.yaml:1]: Hi @FraPazGal,
Thanks for this catch. I used the internet archive / wayback machine and I think I found all the correct redirects that kubernetes.io had in place for these for a time.

**Comment by bcdady** [comment]
RE: "Error: Detected changes in charts without version bump in Chart.yaml" -- perhaps these CI jobs could be updated to ignore changes in comments or description fields, since these would not materially change the behavior of a chart. Such is the case with this PR, where only URLs used in comments or in description fields in the chart are being fixed.

**Comment by bcdady** [comment]
> RE: "Error: Detected changes in charts without version bump in Chart.yaml" -- perhaps these CI jobs could be updated to ignore changes in comments or description fields, since these would not materially change the behavior of a chart. Such is the case with this PR, where only URLs used in comments or in description fields in the chart are being fixed.
---
I suppose the maintainers would prefer to preserve the current chart versions for this docs/links updates, but if that's not accurate, then I'd welcome advise on any good practices for updating many different charts at once (or perhaps this needs to be broken into multiple PRs (per chart? 🥴 I hope that's not necesarry).

File: bitnami/appsmith/values.yaml
Comment (bcdady) [reply bitnami/appsmith/values.yaml:1]: Hi @FraPazGal -- Can we resolve this conversation now?

**Comment by FraPazGal** [comment]
Hi @bcdady, as you said there is no need neither to separate this PR in smaller, per chart, ones nor to bump the chart's versions. Just merging this PR with only its URL changes in the `values.yaml` should suffice.

That said, I think we could use your PR to fix other broken `./user-guide/` links I've encountered, wdyt?

```diff
--- https://kubernetes.io/docs/user-guide/compute-resources/
+++ https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

--- https://kubernetes.io/docs/user-guide/node-selection/
+++ https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/

--- https://kubernetes.io/docs/user-guide/services/
+++ https://kubernetes.io/docs/concepts/services-networking/service

--- https://kubernetes.io/docs/user-guide/ingress/
+++ https://kubernetes.io/docs/concepts/services-networking/ingress/

--- https://kubernetes.io/docs/user-guide/persistent-volumes/
+++ https://kubernetes.io/docs/concepts/storage/persistent-volumes/

--- https://kubernetes.io/docs/user-guide/services/#type-loadbalancer
+++ https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer

--- https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md
+++ https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md
```

**Comment by bcdady** [comment]
I saw that these link changes are still needed, and I'm happy to resume this PR.
Would anyone feel strongly that this work should proceed in a new PR, instead of reopening this one?

**Comment by carrodher** [comment]
Thanks, I think it's better to create a PR from scratch to avoid a lot of conflicts and difficult rebases

**Comment by bcdady** [comment]
Thanks for that feedback @carrodher. In case it helps put your concerns at ease, I've already rebased and cleaned up the new branch. I've just re-reviewed and made sure to include the fixes recommended by @FraPazGal ([here](https://github.com/bitnami/charts/pull/18799#issuecomment-1708180198)).

I'm fine with starting a fresh PR, and just borrowing the description from this one.

**Comment by carrodher** [comment]
I can't reopen this one, not sure if it's because the branch is not part of this repo anymore or something like that. If you are able to recreate it, it's fine, if not there is not any issue in creating a new one

**Pull request closed without merging**